### PR TITLE
Refactor: Create #wait_until_displayed - standardise #displayed? method

### PR DESCRIPTION
**Pull request by TheSpartan1980**
Moved the majority of the logic to the `waiter`
Ensured all specs are still passing
Looked for the opportunity to write additional specs, however it seems the currents specs more than cater for the code.

### CODE REVIEW

**Comment by TheSpartan1980** [comment]
A BIG thank you! to @luke-hill for his patience and guidance

File: lib/site_prism/waiter.rb [hunk 1]
```diff
@@ -14,5 +14,13 @@ def self.wait_until_true(wait_time = Capybara.default_max_wait_time)
 
       raise SitePrism::TimeoutError, "Timed out after #{wait_time}s."
     end
+
+    def self.wait_until_displayed(*args, url)
```
Comment (luke-hill) [inline lib/site_prism/waiter.rb]: This method should essentially house 95% of the logic. And doesn't need to leak state from the original method call. Remember both of these are essentially auto-loaded in

File: lib/site_prism/waiter.rb [hunk 2]
```diff
@@ -14,5 +14,13 @@ def self.wait_until_true(wait_time = Capybara.default_max_wait_time)
...
+      args.last.is_a?(::Hash) ? args.pop : {}
+      seconds = !args.empty? ? args.first : Capybara.default_max_wait_time
+      raise SitePrism::NoUrlMatcherForPageError unless url
```
Comment (luke-hill) [inline lib/site_prism/waiter.rb]: Lets optimise this method call by making this a proper guard clause on line1 of the call

File: lib/site_prism/waiter.rb [hunk 3]
```diff
@@ -14,5 +14,13 @@ def self.wait_until_true(wait_time = Capybara.default_max_wait_time)
 
       raise SitePrism::TimeoutError, "Timed out after #{wait_time}s."
     end
+
+    def self.wait_until_displayed(*args, url)
+      args.last.is_a?(::Hash) ? args.pop : {}
```
Comment (luke-hill) [inline lib/site_prism/waiter.rb]: lets copy this from the original method

File: lib/site_prism/page.rb [hunk 4]
```diff
@@ -65,11 +65,8 @@ def load(expansion_or_html = {}, &block)
 
     def displayed?(*args)
```
Comment (luke-hill) [inline lib/site_prism/page.rb]: lets rewrite this method as the following.

```
def displayed?(*args)
  Waiter.wait_until_displayed(*args) { url_matches?(expected_mappings) }
rescue SitePrism::TimeoutError
  false
end
```

**Comment by luke-hill** [comment]
@TheSpartan1980 are you able to still tackle this now? Want to maybe get this merged in and cut a v3.2 release as a few people have merged in some new features.

**Comment by TheSpartan1980** [comment]
Hi @luke-hill I think I'm going to park this for now I just can't seem to get my head around it. It seems whenever I make changes pre-existing tests break and then when I refactor it to make the test pass the code is less ideal.

**Comment by luke-hill** [comment]
I'll Cherry-pick your commits and pick it up. As I've got nothing I'm working on currently.

**Pull request closed without merging**